docs/src/content/en/docs/frameworks/assistant-ui.mdx

```diff
@@ -0,0 +1,300 @@
+    agents: { chefAgent },
+    // ... other config
+});
+```
+
+This initializes Mastra and makes the `chefAgent` available for use.
+
+### Modify the Chat API endpoints
+
+The initial bootstrapped Next.js project has a `app/api/chat/route.ts` file that exports a `POST` handler. The initial implementation may look like this:
+
+```typescript copy
+import { openai } from "@ai-sdk/openai";
+import { frontendTools } from "@assistant-ui/react-ai-sdk";
+import { streamText } from "ai";
+
+export const runtime = "edge";
+export const maxDuration = 30;
+
+export async function POST(req: Request) {
+  const { messages, system, tools } = await req.json();
+
+  const result = streamText({
+    model: openai("gpt-4o"),
+    messages,
+    // forward system prompt and tools from the frontend
+    toolCallStreaming: true,
+    system,
+    tools: {
+      ...frontendTools(tools),
+    },
+    onError: console.log,
+  });
+
+  return result.toDataStreamResponse();
+}
+```
+
+Now we need to modify the `POST` handler to use the `chefAgent` instead of this implementation.
+
+```typescript copy
+import { mastra } from "@/mastra";
+
+export async function POST(req: Request) {
+  const { messages } = await req.json();
+
+  const agent = mastra.getAgent("chefAgent");
+
+  const stream = await agent.stream(messages);
+
+  return stream.toDataStreamResponse();
+}
+```
+
+Key changes
+- We import the `mastra` instance we created.
+- We use the `mastra.getAgent("chefAgent")` to get the agent we want to use.
+- We use the `agent.stream(messages)` to get the stream of messages from the agent.
+- We return the stream as a data stream response which is compatible with `assistant-ui`.
+
+### Run the application
+
+You're all set! Start your Next.js development server:
+
+```bash copy
+npm run dev
+```
+
+You should now be able to chat with your agent in the browser.
+
+</Steps>
+
+Congratulations! You have successfully integrated Mastra into your Next.js application using the full-stack approach. Your Assistant UI frontend now communicates with a Mastra agent running in your Next.js backend API route.
+
+## Separate Backend Integration
+
+Run Mastra as a standalone server and connect your Next.js frontend (with Assistant UI) to its API endpoints.
+
+<Steps>
+### Create Standalone Mastra Server
+
+Set up your directory structure. A possible directory structure could look like this:
+
+```plaintext
+mastra-server/
+└── (server files)
+
+nextjs-frontend/
+└── (frontend files)
+```
+
+Bootstrap your Mastra server:
+
+```bash copy
+npx create-mastra@latest
+```
+
+This command will launch an interactive wizard to help you scaffold a new Mastra project, including prompting you for a project name and setting up basic configurations.
+Follow the prompts to create your server project.
+
+You now have a basic Mastra server project ready.
+
+<Callout>
+Ensure that you have set the appropriate environment variables for your LLM provider in the `.env.development` file.
```

Comment:
The recent change to the CLI means we now create a `.env` not a `.env.development`, but `.development` will still work